testing/python/raises.py

```diff
@@ -278,3 +278,9 @@ def __class__(self):
                 with pytest.raises(CrappyClass()):
                     pass
             assert "via __class__" in excinfo.value.args[0]
+
+    def test_unicode_message(self):
+        """pytest.raises should be able to match unicode messages when using a unicode regex (#5478)
+        """
+        with pytest.raises(AssertionError, match=u"\u2603"):
+            assert False, u"\u2603"
```

Comment:
can you also test a failure of unicode<=>unicode as well?

Reply:
Sure, but what do you mean by `unicode<=>unicode`? 🤔

Reply:
oh yeah that is super unclear, what the heck did I mean 🤔 

there's 6(?) combinations I think and maybe need more tests for them:

- bytes exception bytes regex success
- bytes exception bytes regex failure
- unicode exception unicode regex success
- unicode exception unicode regex failure (I suspect that this currently causes an undisplayable exception in python2 because *pytest* would be raising a unicode exception)
- ascii unicode exception ascii unicode bytes success (only in py2 due to implicit py2 conversion)
- ascii unicode exception ascii unicode bytes failure (only in py2 due to implicit py2 conversion)

Reply:
I see, thanks. Done.

Surprisingly on Python 3 we don't support raising exceptions using bytes (`raise RuntimeError(b"foo")`), which I find reasonable actually.